### PR TITLE
Update contributor guidelines

### PR DESCRIPTION
**Pull request by cmyr**
This is intended to be a fairly comprehensive contributors guide. Feedback very welcome.

If you've been given commit rights to a repo, please check out the 'review process' section, and if anything is unclear let me know; this represents a change to our current process. In particular, if you have been regularly contributing to one of xi-editor/xi-mac, you will now be expected to merge changes on your own, after they have been approved; and you will be able to seek approval from a larger number of other contributors.

If this gets merged, I will update the xi-mac CONTRIBUTORS file to link to this one.

[Rendered version](https://github.com/xi-editor/xi-editor/blob/3433601c91796244aba0a6eba4aa75858a249ca3/CONTRIBUTING.md)

cc @raphlinus @scholtzan @betterclever @jansol @nangtrongvuon @mqzry @lukakerr @ryanbloom @LiHRaM

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,47 +1,219 @@
...
+- believe the change generally matches the idioms, formatting rules,
+and overall coding style of the relevant repo
+- are ready and able to help resolve any problems that may be introduced by
+merging the change.
+
+If a PR is made by a contributor who has write access to the repo in question, 
+they are responsible for merging/rebasing the PR after it has been approved; 
+otherwise it will be merged by the reviewer.
+
+if a patch adds or modifies behaviour that is observable in the client,
```
Comment (scholtzan) [inline CONTRIBUTING.md]: Small typo → if to If

**Comment by scholtzan** [comment]
This looks good to me! The changes in the review process sound reasonable and like a good improvement to make the process more efficient.

**Comment by jansol** [comment]
Content-wise this looks good to me, but it's quite a wall of text. It might be worth extracting some key points from it into issue/PR templates.

**Comment by mqzry** [comment]
This looks good to me and is a good improvement. Thanks for the effort @cmyr.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,47 +1,219 @@
...
+
+If you are participating in the xi-editor project, you may receive
+additional privileges:
+
+_Organization membership_: If you are regularly making contributions
+to a xi project, in any of the forms outlined above, we will be happy to
+add you to the xi-editor organization, which will give you the ability
+to do things like add labels to issues and view active projects.
+
+_Contributor_: If you are regularly making substantive contributions
```
Comment (ryanbloom) [inline CONTRIBUTING.md]: GitHub gives a "contributor" badge to everyone as soon as their first pull request is merged. This paragraph sounds more like it's describing a "collaborator".

Comment (cmyr) [reply CONTRIBUTING.md]: Yea, I struggled with picking these. 'Collaborator' is also a term that github uses for people who have commit rights on a team but aren't _members_ of that team.

Maybe _Committer_ ?

Comment (raphlinus) [reply CONTRIBUTING.md]: +1 on "Committer". I also like this and feel having lots of detailed guidance will be helpful especially for now contributors.

Comment (ryanbloom) [reply CONTRIBUTING.md]: What was the decision on this? It looks like the "contributor" version got merged.

**Pull request merged**